internal/gps/source.go

```diff
@@ -548,7 +552,7 @@ type source interface {
 	updateLocal(context.Context) error
 	listVersions(context.Context) ([]PairedVersion, error)
 	getManifestAndLock(context.Context, ProjectRoot, Revision, ProjectAnalyzer) (Manifest, Lock, error)
-	listPackages(context.Context, ProjectRoot, Revision) (pkgtree.PackageTree, error)
+	listPackages(context.Context, string, ProjectRoot, Revision) (pkgtree.PackageTree, error)
```

Comment:
this strikes me as unnecessary - it's a fairly straightforward manipulation to select a subtree of a `PackageTree`. as such, this could just be a method on `PackageTree` itself; we needn't push it into the `SourceManager`'s API.

`GetManifestAndLock()` is the harder one that may actually need this.